src/theme-default/logic/copyCode.ts

```diff
@@ -0,0 +1,76 @@
+import { inBrowser } from '../../shared/utils';
+
+export function setupCopyCodeButton() {
+  if (inBrowser()) {
+    const timeoutIdMap: Map<HTMLElement, NodeJS.Timeout> = new Map();
+    window.addEventListener('click', (e) => {
+      const el = e.target as HTMLElement;
+
+      if (el.matches('div[class*="language-"] > button.copy')) {
+        const parent = el.parentElement;
+        const sibling = el.nextElementSibling
+          ?.nextElementSibling as HTMLPreElement | null;
+        if (!parent || !sibling) {
+          return;
+        }
+
+        const { innerText: text = '' } = sibling;
+
+        copyToClipboard(text).then(() => {
+          el.classList.add('copied');
+          clearTimeout(timeoutIdMap.get(el));
+          const timeoutId = setTimeout(() => {
+            el.classList.remove('copied');
+            el.blur();
+            timeoutIdMap.delete(el);
+          }, 2000);
+          timeoutIdMap.set(el, timeoutId);
+        });
+      }
+    });
+  }
+}
+
+async function copyToClipboard(text: string) {
```

Comment:
For `copyToClipboard` function,  I didn't recommnend the implement in VitePress, because we can use the npm library `copy-to-clipboard`, which is really small and statble:

![Uploading image.png…]()

Reply:
Okay, I'll use it instead